Userland/Libraries/LibAudio/WavLoader.cpp

```diff
@@ -293,42 +254,28 @@ MaybeLoaderError WavLoaderPlugin::parse_header()
         }
     }
 
-    ok = ok && (block_size_bytes == (m_num_channels * (bits_per_sample / 8)));
-    CHECK_OK(LoaderError::Category::Format, "Block size sanity check");
+    CHECK(block_size_bytes == (m_num_channels * (bits_per_sample / 8)), LoaderError::Category::Format, "Block size invalid");
 
     dbgln_if(AWAVLOADER_DEBUG, "WAV format {} at {} bit, {} channels, rate {}Hz ",
         sample_format_name(m_sample_format), pcm_bits_per_sample(m_sample_format), m_num_channels, m_sample_rate);
 
-    // Read chunks until we find DATA
+    // Read all chunks before DATA.
     bool found_data = false;
-    u32 data_size = 0;
-    u8 search_byte = 0;
-    while (true) {
-        search_byte = TRY(read_u8());
-        CHECK_OK(LoaderError::Category::IO, "Reading byte searching for data");
-        if (search_byte != 0x64) // D
-            continue;
-
-        search_byte = TRY(read_u8());
-        CHECK_OK(LoaderError::Category::IO, "Reading next byte searching for data");
-        if (search_byte != 0x61) // A
-            continue;
-
-        u16 search_remaining = TRY(read_u16());
-        CHECK_OK(LoaderError::Category::IO, "Reading remaining bytes searching for data");
-        if (search_remaining != 0x6174) // TA
-            continue;
-
-        data_size = TRY(read_u32());
-        found_data = true;
-        break;
+    while (!found_data) {
+        auto chunk_header = TRY(m_stream->read_value<RIFF::ChunkID>());
+        if (chunk_header == RIFF::data_chunk_id) {
+            found_data = true;
+        } else {
+            TRY(m_stream->seek(-RIFF::chunk_id_size, SeekMode::FromCurrentPosition));
+            auto chunk = TRY(m_stream->read_value<RIFF::Chunk>());
+            dbgln("Unhandled WAV chunk of type {}, size {} bytes", chunk.id.as_ascii_string(), chunk.size);
```

Comment:
This might be a bit spammy

Reply:
We have the same for FLAC, and there's usually 3 at maximum per file (I only saw 1)

Reply:
...then I missed it in the FLAC PR :^) maybe make this a proper `dbgln_if(...)`?